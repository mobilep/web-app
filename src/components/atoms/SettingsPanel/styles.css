@media (max-width: 768px) {
	.SettingsPanel {
		--SettingsPanel-aFontSize: inherit;
		--SettingsPanel-aTextDecoration: underline;
		--SettingsPanel-contentMargin: 10px;
		--SettingsPanel-h1Margin: 2em 0 .25em;
		--SettingsPanel-pFontSize: .86em;
		--SettingsPanel-svgMargin: 2.2em 10px 0;
		--SettingsPanel-svgSize: 40px;

		grid-template-areas: 'hr hr' 'icon .';
		grid-template-rows: .5em auto;
		grid-template-columns: calc(var(--SettingsPanel-svgSize) + 1em) auto;
	}
}

@media (min-width: 769px) {
	.SettingsPanel {
		--SettingsPanel-aFontSize: 1.15em;
		--SettingsPanel-aTextDecoration: none;
		--SettingsPanel-contentMargin: 20px;
		--SettingsPanel-h1Margin: 1em 0 .8em;
		--SettingsPanel-pFontSize: inherit;
		--SettingsPanel-svgMargin: 0 20px;
		--SettingsPanel-svgSize: 64px;

		grid-template-areas: 'icon' 'hr' '.';
		grid-template-rows: calc(var(--SettingsPanel-svgSize) + 1em) .5em auto;
	}
}

.SettingsPanel {
	font-size: 14px;
	display: grid;
}

.SettingsPanel svg {
	color: var(--tomatoRed);
	grid-area: icon;
	height: var(--SettingsPanel-svgSize);
	margin: var(--SettingsPanel-svgMargin);
	width: var(--SettingsPanel-svgSize);
}

.SettingsPanel hr {
	background-color: var(--grey2);
	border: 0;
	grid-area: hr;
	height: 1px;
	width: 100%;
}

.SettingsPanel-content {
	margin: 0 var(--SettingsPanel-contentMargin);
}

.SettingsPanel-content h1 {
	color: var(--grey3);
	font-size: inherit;
	font-weight: 400;
	margin: var(--SettingsPanel-h1Margin);
	text-transform: uppercase;
}

.SettingsPanel-content p {
	font-size: var(--SettingsPanel-pFontSize);
	margin-bottom: .3em;
}

.SettingsPanel-content button,
.SettingsPanel-content select {
	margin-top: .5em;
	text-align: inherit;
}

.SettingsPanel-content a {
	color: var(--tomatoRed);
	font-size: var(--SettingsPanel-aFontSize);
	text-decoration: var(--SettingsPanel-aTextDecoration);
}

.SettingsPanel-content button {
	font-size: var(--SettingsPanel-aFontSize);
}

.SettingsPanel-content a:focus,
.SettingsPanel-content a:hover {
	color: var(--tomatoRed);
	text-decoration: underline;
}
