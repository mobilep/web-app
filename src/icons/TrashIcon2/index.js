export { default } from './TrashIcon2';
