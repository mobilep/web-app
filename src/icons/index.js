export { default as ArrowIcon } from './ArrowIcon';
export { default as AttachIcon } from './AttachIcon';
export { default as BellIcon } from './BellIcon';
export { default as CalendarIcon } from './CalendarIcon';
export { default as CheckIcon } from './CheckIcon';
export { default as CirclePlusIcon } from './CirclePlusIcon';
export { default as CloseIcon } from './CloseIcon';
export { default as CloudIcon } from './CloudIcon';
export { default as DotsIcon } from './DotsIcon';
export { default as EmptyInboxIcon } from './EmptyInboxIcon';
export { default as EmptyScenarioIcon } from './EmptyScenarioIcon';
export { default as EmptyTeamIcon } from './EmptyTeamIcon';
export { default as ErrorIcon } from './ErrorIcon';
export { default as EvaluationIcon } from './EvaluationIcon';
export { default as EyeFillIcon } from './EyeFillIcon';
export { default as EyeIcon } from './EyeIcon';
export { default as FileIcon } from './FileIcon';
export { default as FolderIcon } from './FolderIcon';
export { default as GlobeIcon } from './GlobeIcon';
export { default as InboxIcon } from './InboxIcon';
export { default as ImageIcon } from './ImageIcon';
export { default as LogoIcon } from './LogoIcon';
export { default as LogOutIcon } from './LogOutIcon';
export { default as PlayIcon } from './PlayIcon';
export { default as PlusIcon } from './PlusIcon';
export { default as ProfileIcon } from './ProfileIcon';
export { default as QuestionIcon } from './QuestionIcon';
export { default as RecordCameraIcon } from './RecordCameraIcon';
export { default as RerecordIcon } from './RerecordIcon';
export { default as RecordIcon } from './RecordIcon';
export { default as RecordSendIcon } from './RecordSendIcon';
export { default as ReplayIcon } from './ReplayIcon';
export { default as ScenariousIcon } from './ScenariousIcon';
export { default as SearchIcon } from './SearchIcon';
export { default as SettingsIcon } from './SettingsIcon';
export { default as SpinnerIcon } from './SpinnerIcon';
export { default as SuccessIcon } from './SuccessIcon';
export { default as SwitchIcon } from './SwitchIcon';
export { default as SwitchCameraIcon } from './SwitchCameraIcon';
export { default as TeamIcon } from './TeamIcon';
export { default as TrashIcon } from './TrashIcon';
export { default as TrashIcon2 } from './TrashIcon2';
export { default as VideoCameraIcon } from './VideoCameraIcon';
export { default as WarningIcon } from './WarningIcon';
export { default as MicIcon } from './MicIcon';
export { default as PdfIcon } from './PdfIcon';
export { default as MenuOpenedIcon } from './MenuOpenedIcon';
export { default as LogoRedIcon } from './LogoRedIcon';
export { default as GroupChatIcon } from './GroupChatIcon';
export { default as StarIconGold } from './StarIconGold';
export { default as StarIcon } from './StarIcon';
export { default as ArrowForwardIcon } from './ArrowForwardIcon';
