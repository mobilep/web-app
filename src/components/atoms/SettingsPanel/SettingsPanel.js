import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './styles.css';

export default class SettingsPanel extends Component {

	static propTypes = {
		action: PropTypes.oneOfType([
			PropTypes.element,
			PropTypes.arrayOf(PropTypes.element),
		]),
		description: PropTypes.string,
		icon: PropTypes.element.isRequired,
		title: PropTypes.string.isRequired,
	}

	render () {
		const { action, description, icon, title } = this.props;
		const desc = description ? (<p>{description}</p>) : null;

		return (
			<section className='SettingsPanel'>
				{icon}
				<hr />
				<div className='SettingsPanel-content'>
					<h1>{title}</h1>
					{desc}
					{action}
				</div>
			</section>
		);
	}

}
