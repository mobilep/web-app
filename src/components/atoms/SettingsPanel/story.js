import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs/react';
import { QuestionIcon } from '../../../icons';

import SettingsPanel from './';

storiesOf('1. Atoms / SettingsPanel', module)
	.add('SettingsPanel', () => (
		<SettingsPanel
			title={text('Title', 'Help')}
			description={text('Description', 'Click the link below for fast help:')}
			icon={<QuestionIcon />}
			action={<a href='#' target='_blank'>help.mobilepractice.io</a>}
		/>
	));
