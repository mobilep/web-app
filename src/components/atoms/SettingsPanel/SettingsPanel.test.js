import React from 'react';
import { shallow } from 'enzyme';
import { QuestionIcon } from '../../../icons';

import SettingsPanel from './SettingsPanel';

describe('Atom - SettingsPanel', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<SettingsPanel
				title='Title'
				description='Description'
				icon={<QuestionIcon />}
				action={<a href='#'>Some link</a>}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

});
