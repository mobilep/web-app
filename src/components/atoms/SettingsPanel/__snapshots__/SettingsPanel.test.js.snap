// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - SettingsPanel Should render 1`] = `
<section
  className="SettingsPanel"
>
  <QuestionIcon />
  <hr />
  <div
    className="SettingsPanel-content"
  >
    <h1>
      Title
    </h1>
    <p>
      Description
    </p>
    <a
      href="#"
    >
      Some link
    </a>
  </div>
</section>
`;
